VespaGAS/Controllers/GasCalculatorViewController.swift

```diff
@@ -12,25 +12,25 @@ class GasCalculatorViewController: UIViewController {
 
     var viewModel: GasCalculatorViewModel? = nil
     let disposeBag = DisposeBag()
+    let store = VespaStore.sharedInstance
 
     override func viewDidLoad() {
-        let gasolineObservable = self.gasSlider.rx_value.asObservable().map { (amount) -> Gasoline in
-            return Gasoline(amount: amount, unit: .Liter)
+        _ = self.gasSlider.rx_value.asObservable().subscribeNext { (amount) in
+            self.store.dispatch(UpdateWithGasolineAction(gasoline: Gasoline(amount: amount, unit: .Liter)))
```

Comment:
Every-time we want to update our state, we have to dispatch an action to the store